docs: explain specific features added on top of Visual Studio Code (#2505)
codenvy/che-docs: modules/end-user-guide/pages/developer-workspaces.adoc

* Tool dependencies needed by developers to work on a project
* Application runtime: a replica of the environment where the application runs in production

Pods manage each component of a {prod-short} workspace.
Therefore, everything running in a {prod-short} workspace is running inside containers.

> This makes a {prod-short} workspace highly portable.

Reviewer: I don't think splitting the text into separate lines was necessary. It breaks the flow when I read it.

Author: It renders as a single paragraph.

> The embedded browser-based IDE is the point of access for everything running in a {prod-short} workspace.
> This makes a {prod-short} workspace easy to share.

Reviewer: ```suggestion
The embedded browser-based IDE is the point of access for everything running in a {prod-short} workspace. This makes a {prod-short} workspace easy to share.
```

Visual Studio Code is the default browser-based IDE. {prod-short} adds these features:

Tasks::

> In Visual Studio Code, commands that are defined in the devfile appear as tasks.

Reviewer: ```suggestion
Commands that are defined in the devfile appear as tasks.
```

+

> To execute such command, go to menu:Terminal[Run Task].

Reviewer: ```suggestion
To execute such a command, go to menu:Terminal[Run Task].
```

+
Alternatively, kbd:[F1] to open the *Command Palette* and type `Tasks: Run Tasks` to see a drop-down list of available task-related commands.

> Process monitoring::

Reviewer 2: There are 2 distinct VS Code built-in extensions: one that "monitors resources (CPU and RAM)", and another one "exposes ports". You are talking about the second one here but it can be confused with the resource monitoring.

> When a process starts listening to a port, {prod-short} displays an alert with a link to open the resulting resource.

Reviewer 2: I would say 'notification' rather then 'alert'. And there are actually 2 notifications: one prompts the user to expose the port externally or not. The second with a link to open the resulting resource.

* Tool dependencies needed by developers to work on a project
* Application runtime: a replica of the environment where the application runs in production

Pods manage each component of a {prod-short} workspace.
Therefore, everything running in a {prod-short} workspace is running inside containers.
This makes a {prod-short} workspace highly portable.

The embedded browser-based IDE is the point of access for everything running in a {prod-short} workspace.

> == Visual Studio Code

Reviewer: ```suggestion
== Microsoft Visual Studio Code - Open Source
```

> Visual Studio Code is the default browser-based IDE.

Reviewer: ```suggestion
Microsoft Visual Studio Code - Open Source is the default browser-based IDE.
```

{prod-short} adds these extensions:

Commands::

> Translates Devfile commands to Visual Studio Code tasks.

Reviewer: ```suggestion
Translates Devfile commands to Microsoft Visual Studio Code - Open Source tasks.
```

+
.Procedure
* To see the drop-down list of available tasks, type: kbd:[F1] `Tasks: Run Task` kbd:[Enter] `che`.

Activity tracker::

> Tracks events provided by the Visual Studio Code to determine and stop inactive workspaces.

Reviewer: ```suggestion
Tracks events provided by the Microsoft Visual Studio Code - Open Source to determine and stop inactive workspaces.
```

This extension does not save, collect, or store data.

API::
Provides helpers to interact with {devworkspace} and {prod-short}.

GitHub authentication::
Provides support for authenticating to GitHub.
It registers the `github` Authentication Provider that can be leveraged by other extensions.
This also provides the GitHub authentication used by Settings Sync.

Port::
Detects opening ports and provides redirect URI.

> When a process starts listening to a port, {prod-short} displays an alert with a link to open the resulting resource.

Author: ```suggestion
When a process starts listening to a port, {prod-short} displays a notification with a link to open the resulting resource.
```

+
.Procedure
* To display the endpoint list, type: kbd:[F1] `Explorer: Focus on endpoints View` kbd:[Enter].

Remote::
Provides commands for the remote authority.

Resource monitor::
Monitors resources such as CPU and RAM.

Telemetry::

> Detects and sends the following events to a backend telemetry plugin listening on `http://localhost:$\{DEVWORKSPACE_TELEMETRY_BACKEND_PORT}`:

Author: ```suggestion
Detects and sends the following events to a backend telemetry plugin listening on `\http://localhost:$\{DEVWORKSPACE_TELEMETRY_BACKEND_PORT}`:
```
